tests/plots/test_summary_plot.py

```diff
@@ -34,3 +35,31 @@ def test_summary_plot(explainer):
     shap.plots.beeswarm(shap_values, show=False)
     plt.tight_layout()
     return fig
+
+
+@pytest.mark.parametrize(
+    "rng",
+    [
+        np.random.default_rng(167089660),
+        17,
+        np.random.SeedSequence(entropy=60767),
+    ],
+)
+def test_summary_plot_seed_insulated(explainer, rng):
+    # ensure that it is possible for downstream
+    # projects to avoid mutating global NumPy
+    # random state
+    # see i.e., https://scientific-python.org/specs/spec-0007/
+    shap_values = explainer(explainer.data)
+    state_before = np.random.get_state()[1]  # type: ignore[index]
+    shap.summary_plot(shap_values, show=False, rng=rng)
+    state_after = np.random.get_state()[1]  # type: ignore[index]
+    assert_array_equal(state_after, state_before)
+
+
+def test_summary_plot_warning(explainer):
+    # enforce FutureWarning for usage of global random
+    # state as we prepare for SPEC 7 adoption
+    shap_values = explainer(explainer.data)
+    with pytest.warns(FutureWarning, match="NumPy global RNG"):
+        shap.summary_plot(shap_values, show=False)
```

Comment:
The `FutureWarning `is correctly issued here it seems. I did wonder if it was not ideal that we may be depending on the fact that (I think) the `shap` testsuite as a whole does a bunch of global random seed setting. If that gets cleaned up at some point I suppose this test might need to be adjusted, but at least the test should be noisy about it, and by then we might be closer to ready to avoid global seed usage anyway.